tests/RuleSetTest.php

```diff
@@ -306,19 +306,54 @@ public function provideSafeSets()
             array(array('@PSR1' => true), true),
             array(array('@PSR2' => true), true),
             array(array('@Symfony' => true), true),
+            array(array('@Symfony:risky' => true), false),
+            array(array('@PHP56Migration:risky' => true), false),
             array(
                 array(
                     '@Symfony:risky' => true,
                     '@Symfony' => false,
                 ),
                 false,
             ),
-            array(
-                array(
-                    '@Symfony:risky' => true,
-                ),
-                false,
-            ),
+        );
+    }
+
+    /**
+     * @param array $set
+     * @param bool  $safe
+     *
+     * @dataProvider providePhp70SafeSets
+     * @requires PHP 7.0
+     */
+    public function testPhp70MigrationSet(array $set, $safe)
+    {
+        $this->testRiskyRulesInSet($set, $safe);
+    }
+
+    public function providePhp70SafeSets()
+    {
+        return array(
+            array(array('@PHP70Migration' => true), true),
+            array(array('@PHP70Migration:risky' => true), false),
+        );
+    }
+
+    /**
+     * @param array $set
+     * @param bool  $safe
+     *
+     * @dataProvider providePhp71SafeSets
+     * @requires PHP 7.1
+     */
+    public function testPhp71MigrationSet(array $set, $safe)
+    {
+        $this->testRiskyRulesInSet($set, $safe);
```

Comment:
to be honest I'm not sure if calling real test method from inside of other test method is a good idea

Reply:
I wasn't either, but then I've found 6 cases of this in tests e.g. [here](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.5/tests/Fixer/DoctrineAnnotation/DoctrineAnnotationSpacesFixerTest.php#L461). Want me to change to doTest-solution?

One more question: can I create `@PHP71Migration:risky` group (which will only has `@PHP70Migration:risky` in 2.2 branch)? It's not against YAGNI as we know we are going to need it.

Reply:
I've no strong opinion about point 1.

About point two; maybe we can have a meta test that checks that [all know set-names](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.5/src/RuleSet.php#L270) are being tested by checking the test providers? Than missing `@PHP71Migration:risky` test on higher branch will show up during merging. It would also protect us in the future when adding a new set, WDYT?

Reply:
:+1: for 1st of Kuba's paragraph
:+1: for 2nd paragraph of @SpacePossum 
(thus :-1: for 2nd paragraph from Kuba, no need for a ruleset that would be the same as other one - same way we don't have ruleset for 7.2)